cla-backend-go/v2/company/service.go

```diff
@@ -0,0 +1,174 @@
+package company
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/communitybridge/easycla/cla-backend-go/utils"
+
+	"github.com/aws/aws-sdk-go/aws"
+	v1Models "github.com/communitybridge/easycla/cla-backend-go/gen/models"
+	v1SignatureParams "github.com/communitybridge/easycla/cla-backend-go/gen/restapi/operations/signatures"
+	"github.com/communitybridge/easycla/cla-backend-go/gen/v2/models"
+	"github.com/communitybridge/easycla/cla-backend-go/logging"
+	"github.com/communitybridge/easycla/cla-backend-go/signatures"
+	v2UserService "github.com/communitybridge/easycla/cla-backend-go/v2/user-service"
+	v2UserServiceModels "github.com/communitybridge/easycla/cla-backend-go/v2/user-service/models"
+)
+
+// Service functions for company
+type Service interface {
+	GetCompanyCLAManagers(companyID string) (*models.CompanyClaManagers, error)
+}
+
+// ProjectRepo contains project repo methods
+type ProjectRepo interface {
+	GetProjectByID(projectID string) (*v1Models.Project, error)
+}
+
+type service struct {
+	signatureRepo signatures.SignatureRepository
+	projectRepo   ProjectRepo
+}
+
+// NewService returns instance of company service
+func NewService(sigRepo signatures.SignatureRepository, projectRepo ProjectRepo) Service {
+	return &service{
+		signatureRepo: sigRepo,
+		projectRepo:   projectRepo,
+	}
+}
+
+func (s *service) getAllCCLASignatures(companyID string) ([]v1Models.Signature, error) {
+	var sigs []v1Models.Signature
+	var lastScannedKey *string
+	for {
+		signatures, err := s.signatureRepo.GetCompanySignatures(v1SignatureParams.GetCompanySignaturesParams{
+			CompanyID:     companyID,
+			SignatureType: aws.String("ccla"),
+			NextKey:       lastScannedKey,
+		}, 1000)
+		if err != nil {
+			return nil, err
+		}
+		sigs = append(sigs, signatures.Signatures...)
+		if signatures.LastKeyScanned == "" {
+			break
+		}
+		lastScannedKey = aws.String(signatures.LastKeyScanned)
+	}
+	return sigs, nil
+}
+
+func (s *service) GetCompanyCLAManagers(companyID string) (*models.CompanyClaManagers, error) {
+	sigs, err := s.getAllCCLASignatures(companyID)
+	if err != nil {
+		return nil, err
+	}
+	var claManagers []*models.CompanyClaManager
+	lfUsernames := utils.NewStringSet()
+	projectIDs := utils.NewStringSet()
+	// Get CLA managers
+	for _, sig := range sigs {
+		for _, user := range sig.SignatureACL {
+			claManagers = append(claManagers, &models.CompanyClaManager{
+				ApprovedOn: "",
```

Comment:
Just make a note/comment here - DB currently doesn't store the approved on value.

Reply:
Done